src/preprocess/run_tg.py

```diff
@@ -43,105 +66,143 @@
     #           # scatter_plot_path=os.path.join(TG_OUT, 'global_pca.html'),
     #           bin_file_type='--bfile')
 
+
     # 1. Centralised variant QC
-    logger.info("Centralised variant QC")
     varqc_prefix = TG_BFILE_PATH + '_varqc'
-    QC.qc(input_prefix=TG_BFILE_PATH, output_prefix=varqc_prefix, qc_config=variant_qc_config)
+    if Stage.VARIANT_QC in stages:
+        logger.info('Centralised variant QC')
+        QC.qc(input_prefix=TG_BFILE_PATH, output_prefix=varqc_prefix, qc_config=variant_qc_config)
 
     # 2. Split into ethnic datasets and then QC each local dataset
-    logger.info("Splitting ethnic dataset")
-    nodes = SplitTGHeter().split(input_prefix=varqc_prefix, make_pgen=True)
+    nodes = set(TG_SUPERPOP_DICT.values())
+    if Stage.POPULATION_SPLIT in stages:
+        logger.info('Splitting ethnic dataset')
+        SplitTGHeter().split(input_prefix=varqc_prefix, make_pgen=True)
 
     # 3. Perform sample QC on each node separately
-    for local_prefix in nodes:
-        logger.info(f'Running local QC for {local_prefix}')
-        local_samqc_prefix = os.path.join(SPLIT_GENO_DIR, local_prefix) + '_filtered'
-        QC.qc(input_prefix=os.path.join(SPLIT_GENO_DIR, local_prefix), output_prefix=local_samqc_prefix, qc_config=sample_qc_config)
-
-    # # 4. Perform pruning for each node separately
-    # pruning = Pruning()
-    # for node in nodes:
-    #     logger.info(f'Pruning for {node}')
-    #     pruning.prune(node + '_filtered', **default_pruning)
-    # logger.info('Merging nodes from TG_SUPERPOP_DICT')
-    # pruning.merge()
-    # for node in nodes:
-    #     logger.info(f'Removing variants from {node}')
-    #     pruning.remove_variants(NODE=f'{node}_filtered', SNPS='ALL')
+    if Stage.SAMPLE_QC in stages:
+        for local_prefix in nodes:
+            logger.info(f'Running local QC for {local_prefix}')
+            local_samqc_prefix = os.path.join(SPLIT_GENO_DIR, local_prefix) + '_filtered'
+            QC.qc(
+                input_prefix=os.path.join(SPLIT_GENO_DIR, local_prefix),
+                output_prefix=local_samqc_prefix,
+                qc_config=sample_qc_config
+            )
 
+    # 4. Perform pruning for each node separately
+    if Stage.PRUNING in stages:
+        logger.info(f'Pruning with plink')
+        PlinkPruningRunner(
+            source_directory=SPLIT_GENO_DIR,
+            nodes=nodes,
```

Comment:
Because `ALL` is now in `nodes`, it needs to be removed from the list for pruning.
```suggestion
            nodes=nodes_noall,
```